join.go

```diff
@@ -580,6 +591,35 @@ func (js *joinset) JoinIntoBatch() (models.Batch, bool) {
 	return newBatch, true
 }
 
+// outName returns the fully qualified name for field k, in stream joined stream i.
+//
+// If the specified prefix is empty, the output name is k.
+//
+// If the specified prefix contains a trailing #, the output name is the prefix
+// without the trailing # concatenated with k
+//
+// Otherwise, the output name is prefix+"."+k
+//
+// A runtime check is performed to guarantee that all output points have unique field
+// names.
+func (js *joinset) outName(i int, k string) string {
+	prefix := js.prefixes[i]
+	if len(prefix) == 0 {
+		prefix = ""
+	} else if prefix[len(prefix)-1] == '#' {
+		prefix = prefix[0 : len(prefix)-1]
+	} else {
+		prefix = prefix + "."
+	}
+	n := prefix + k
+	if claim, ok := js.claims[n]; ok && claim != i {
+		panic(fmt.Errorf("field %s of input %d conflicts with input %d", k, i, claim))
```

Comment:
Could we remove the panic here and return an error instead?

The calling functions `JoinInto*` can then log the error and `return models.Point/Batch{},false` the join will be skipped.